pkg/expression/builtin_string.go

```diff
@@ -3786,7 +3875,36 @@ func (b *builtinInstrUTF8Sig) evalInt(ctx EvalContext, row chunk.Row) (int64, bo
 		substr = strings.ToLower(substr)
 	}
 
-	idx := strings.Index(str, substr)
+	position := int64(1)
+	if len(b.args) >= 3 {
+		position, IsNull, err = b.args[2].EvalInt(ctx, row)
+		if IsNull || err != nil {
+			return 0, true, err
+		}
+	}
+
+	if position == 0 {
+		return int64(0), false, nil
+	}
+
+	occurrences := int64(1)
+	if len(b.args) >= 4 {
+		occurrences, IsNull, err = b.args[3].EvalInt(ctx, row)
+		if IsNull || err != nil {
+			return 0, true, err
+		}
+		if occurrences <= 0 {
+			return 0, true, errors.Errorf("Occurrence should not be negative, got %d", occurrences)
+		}
+	}
+
+	var idx int
+	if position > 0 {
+		idx = findNthOccurrenceUTF8(str, substr, int(position-1), int(occurrences))
+	} else {
+		idx = findLastOccurrenceUTF8(str, substr, int(-position), int(occurrences))
+	}
```

Comment:
why we need to search `substr` using different way? Seems if `position` is negative will only search from `str[-position]`🤔

Reply:
Because when `position` is negative, search is start from `-position` and end at `0`, not from `-position` to `len(str) - 1`.